packages/app/app/components/Dashboard/BestNewMusicTab/BestNewMusicMenu/BestNewMusicCard/index.tsx

```diff
@@ -32,20 +61,24 @@ const FavoriteIcon = ({ isFavorite, onClick, dataTestId }) =>
     onClick={onClick}
   />;
 
-const BestNewMusicCard = ({
-  actions,
+const BestNewMusicCard: React.FC<BestNewMusicCardProps> = ({
   item,
-  favoriteTrack,
   onClick,
-  settings,
-  withFavoriteButton
+  withFavoriteButton,
+  favoriteTrack,
+  actions,
+  settings
 }) => {
   const { artist, title, thumbnail } = item;
 
+  const handleClick = () => {
```

Comment:
Added this click handling function to satisfy the event handler error typescript would raise with the previous onClick implementation

Reply:
The problem with typescript and the previous implementation was that `setActiveItem`, when used as an onClick callback, would require the method (declared in BestNewMusicTab) to have a MouseEventHandler type as its first argument. `setActiveItem` is also called directly in its home component (BestNewMusicTab - index.tsx line 35), in which case typescript would not be happy about the method having the MouseHandlerEvent (its called from componentDidMount). 

I feel like the `setActiveItem` method shouldn't be concerned with click handlers in its child components, which is why I went for the approach of adding a local click handler to `bestNewMusicCard`. What are your thoughts? 🤔 

I agree about the div conventionality though 🤔 should we change this Div element to a button  / anchor anyway?

Reply:
I think this is an issue with the parent component and how it passes down this prop - since you're only converting JS to TS, let's keep it this way in this PR. It can potentially be fixed in a separate commit in the future.